spec/lib/dfe_sign_in_api_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+require 'dfe_sign_in_api'
+
+RSpec.describe DFESignIn::API do
+  describe '#users' do
+    it 'returns the users from the API ' do
+      stub_api_response_for_page(1)
+      response = described_class.new.users
+
+      expect(response).to eq(JSON.parse(response_file(1)))
+      expect(response['page']).to eq(1)
+      expect(response['numberOfPages']).to eq(2)
+    end
+
+    it 'returns the users from the API for a given page ' do
+      stub_api_response_for_page(2)
+      response = described_class.new.users(page: 2)
+
+      expect(response).to eq(JSON.parse(response_file(2)))
+      expect(response['page']).to eq(2)
+      expect(response['numberOfPages']).to eq(2)
+    end
+
+    it 'sets a token in the header of the request' do
+      jwt_token = double
+      expect(JWT).to receive(:encode).with(
+        {
+          iss: 'schooljobs',
+          exp: (Time.now.getlocal + 60).to_i,
+          aud: 'signin.education.gov.uk'
+        },
+        'test-password',
+        'HS256'
+      ).and_return(jwt_token)
+
+      stub_api_response_with_token(jwt_token)
```

Comment:
Is this required in conjunction with the `expect` below?  Can this line be removed?

Reply:
Refactored this in 935caaf